fenix/app/src/main/java/org/mozilla/fenix/shopping/ReviewQualityCheckFragment.kt

```diff
@@ -30,10 +30,10 @@ class ReviewQualityCheckFragment : BottomSheetDialogFragment() {
     private var behavior: BottomSheetBehavior<View>? = null
     private val store by lazy {
         ReviewQualityCheckStore(
-            reviewQualityCheckPreferences = ReviewQualityCheckPreferencesImpl(
-                requireComponents.settings,
+            middleware = MiddlewareProvider.provideMiddleware(
```

Comment:
The value here is that we are separating the creation of dependencies from the fragment. Ideally, it would be good to have constructor injection with Fragment but the next best thing would be to reduce/remove the coupling of the dependencies from the fragment, here with field injection. The list will keep on increasing as other middlewares are added. Just to add, I think it's not a new pattern, it's just separating the concerns by extracting the injection code and even making the Fragment simpler to read and digest, `EngineMiddleware` is doing something similar. Saying that I do think it's worthwhile. wdyt?

Reply:
Learned about constructor/field injecton here. sgtm